zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -857,6 +862,9 @@ where
                     {
                         Address::Transparent(t) => Some(t),
                         Address::Unified(uaddr) => uaddr.transparent(),
+                        Address::TransparentSourceOnly(_) => {
+                            unreachable!("Transparent-source-only addresses should not occur here")
```

Comment:
This is in fact reachable by deserialising a proposal, and should be an error rather than a panic.